### PR TITLE
Update schema tests

### PR DESCRIPTION
**Pull request by gregarndt**

### CODE REVIEW

File: runtime/compositeschema_test.go [hunk 1]
```diff
@@ -91,5 +88,112 @@ func TestInvalidSchemaReference(t *testing.T) {
...
+
+	composites := make([]CompositeSchema, 4)
+	for i := range composites {
+		// Create unique property names to avoid collision resulting in error
+		propertyName := fmt.Sprintf("prop%d", i)
+		composite, err := NewCompositeSchema(propertyName, schema, true, func() interface{} { return nil })
+		if err != nil {
+			t.Fatal(err)
+		}
+		composites[i] = composite
```
Comment (jonasfj) [inline runtime/compositeschema_test.go]: This is fancy...

Note: typically I would do `composites := []CompositeSchema{}` and then do `composites = append(composites, composite)` as it's safer... Well, not really when you do range over composites... nevermind this is smart :)

File: runtime/compositeschema_test.go [hunk 2]
```diff
@@ -91,5 +88,112 @@ func TestInvalidSchemaReference(t *testing.T) {
...
+	}
+
+	merged, err := MergeCompositeSchemas(composed1, composed2)
+	if err != nil {
+		t.Fatalf("Error creating a merged composite scheme. %v", err)
+	}
 
+	m := merged.(*composedSchema)
+	if len(m.entries) != 2 {
+		t.Fatalf("Merged schema should have 4 entries but had %d", len(m.entries))
```
Comment (jonasfj) [inline runtime/compositeschema_test.go]: nope... does this test pass? You are merging in a tree... So you should only have two... as both of those each has two...

Comment (jonasfj) [inline runtime/compositeschema_test.go]: oh, wait it's only the comment that is out of date :)

Comment (gregarndt) [inline runtime/compositeschema_test.go]: ah yea, sorry, the comment was out of date after I realize it's just two entries

**Pull request merged**